### PR TITLE
adding skip-confirmation flag to move config - fix to 158

### LINKED ISSUE #158: Is there a way to bypass 2to3 confirm ?
we are using the helm 2to3 convert grammatically in c# ,is there a way to bypass this confirmation : 
> helm3 2to3 move config
2020/07/29 10:50:21 WARNING: Helm v3 configuration may be overwritten during this operation.
2020/07/29 10:50:21
[Move Config/confirm] Are you sure you w**ant to move the v2 configuration? [y/N]:**

we are using the c# start process : 
 new Process
                {
                    StartInfo = new ProcessStartInfo
                    {
                        FileName = "helm3",
                        Arguments = $"{arguments}",
                        RedirectStandardOutput = true,
                        RedirectStandardError = true,
                        CreateNoWindow = true,
                        UseShellExecute = false
                    }
                };
what will be the best way to send yes ,or to bypass this message ?
thank you.

### PR DESCRIPTION
**Pull request by digeler**
i opened a new one with just one commit this time ,and closed pr 158 please review.
thank you.

Fixes #158

### CODE REVIEW

File: cmd/move_config.go [hunk 1]
```diff
@@ -66,12 +69,14 @@ func Move(dryRun bool) error {
...
 	log.Println("WARNING: Helm v3 configuration may be overwritten during this operation.")
 	log.Println()
-	doCleanup, err := utils.AskConfirmation("Move Config", "move the v2 configuration")
-	if err != nil {
-		return err
+	if skipConfirmation {
+		log.Println("Skipping confirmation before performing move.")
+		doCleanup = true
+	} else {
+		doCleanup, err = utils.AskConfirmation("Move Config", "move the v2 configuration")
```
Comment (marckhouzam) [inline cmd/move_config.go:76]: The check for `err != nil` is missing here.

File: cmd/move_config.go [hunk 2]
```diff
@@ -55,6 +55,9 @@ func runMove(cmd *cobra.Command, args []string) error {
 	return Move(settings.DryRun)
 }
 
+var err error
+var doCleanup bool
```
Comment (marckhouzam) [inline cmd/move_config.go:59]: These should not be global. How about creating them just before they are used in the function?

File: cmd/move_config.go (hunk 1 above)
Comment (digeler) [reply cmd/move_config.go:76]: thanks added

File: cmd/move_config.go (hunk 2 above)
Comment (digeler) [reply cmd/move_config.go:59]: done also thanks

**Comment by digeler** [comment]
thank you , waiting for the maintainer to approve the pr.

File: cmd/move_config.go [hunk 3]
```diff
@@ -58,6 +58,8 @@ func runMove(cmd *cobra.Command, args []string) error {
 // Moves/copies v2 configuration to v2 configuration. It copies repository config,
 // plugins and starters. It does not copy cache.
 func Move(dryRun bool) error {
+	var err error
+	var doCleanup bool
```
Comment (hickeyma) [inline cmd/move_config.go:62]: doCleanup --> doConfig

File: cmd/move_config.go [hunk 4]
```diff
@@ -66,12 +68,18 @@ func Move(dryRun bool) error {
...
+		doCleanup = true
+	} else {
+		doCleanup, err = utils.AskConfirmation("Move Config", "move the v2 configuration")
+		if err != nil {
+			return err
+		}
 	}
 	if !doCleanup {
-		log.Println("Move configuration will not proceed as the user didn't answer (Y|y) in order to continue.")
+		log.Println("Move will not proceed as the user didn't answer (Y|y) in order to continue.")
```
Comment (hickeyma) [inline cmd/move_config.go:82]: No need to change this

File: cmd/move_config.go [hunk 5]
```diff
@@ -66,12 +68,18 @@ func Move(dryRun bool) error {
 
 	log.Println("WARNING: Helm v3 configuration may be overwritten during this operation.")
 	log.Println()
-	doCleanup, err := utils.AskConfirmation("Move Config", "move the v2 configuration")
-	if err != nil {
-		return err
+	if skipConfirmation {
+
+		log.Println("Skipping confirmation before performing move.")
```
Comment (hickeyma) [inline cmd/move_config.go:73]: move --> move config

**Comment by rimusz** [comment]
@digeler can you also please update the readme with the new flag https://github.com/helm/helm-2to3#migrate-helm-v2-configuration

**Comment by rimusz** [comment]
@digeler please sign DCO

File: cmd/move_config.go (hunk 5 above)
Comment (hickeyma) [reply cmd/move_config.go:73]: This has not been changed.

File: cmd/move_config.go (hunk 4 above)
Comment (hickeyma) [reply cmd/move_config.go:82]: The print statement is still changed

**Comment by digeler** [comment]
Is there something else needed.
What’s holding the merge?

File: cmd/move_config.go (hunk 5 above)
Comment (hickeyma) [reply cmd/move_config.go:73]: Can you change: "Skipping confirmation before performing move." to "Skipping confirmation before performing move configuration."

File: cmd/move_config.go (hunk 4 above)
Comment (hickeyma) [reply cmd/move_config.go:82]: Can you not make this change: `log.Println("Move will not proceed as the user didn't answer (Y|y) in order to continue.")`. In other words, leave the message as it was.

File: README.md [hunk 6]
```diff
@@ -89,6 +89,7 @@ $ helm 2to3 move config [flags]
 Flags:
 
       --dry-run  simulate a command
+       --skip-confirmation   if set, skips confirmation message before performing move
```
Comment (hickeyma) [inline README.md:92]: align `--skip-confirmation` with `--dry-run`

**Review by hickeyma — CHANGES_REQUESTED** [review]
@digeler There are some changes still not resolved. It would be great if you could resolve them.

**Comment by hickeyma** [comment]
@digeler looking forward to getting this merged when you get a chance to update it.

File: README.md (hunk 6 above)
Comment (digeler) [reply README.md]: done

**Comment by digeler** [comment]
Any updates am i missing something ?

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @digeler

**Pull request merged**